udapi/core/node.py

```diff
@@ -366,7 +365,15 @@ def descendants(self):
          nodes4 = [n for n in node.descendants if n.ord < node.ord] + [node]
         See documentation of ListOfNodes for details.
         """
-        return ListOfNodes(self.unordered_descendants(), origin=self)
+        stack = list(self._children)
+        descendants = ListOfNodes(stack, origin=self)
+        while(stack):
+            n = stack.pop()
+            if n._children:
+                stack.extend(n._children)
+                descendants.extend(n._children)
+        descendants.sort()
+        return descendants
```

Comment:
Yes, it does the same as `ListOfNodes(sorted(self.unordered_descendants()), origin=self)`, but faster.
The `unordered_descendants()` method creates a new list of nodes (and Python calls `malloc` under the hood). `sorted` creates another list of nodes (and the original list is garbage-collected if it is not referenced elsewhere). Finally, `x = ListOfNodes(my_list)` creates another list of nodes. I've spent some time trying to reuse the memory of `my_list`, but the only way is to store it in `ListOfNode._data`, but such delegation makes everything else slower, e.g. `x[3]` calls `ListOfNodes.__getitem__` instead of C-optimized native list index.
By using `sort` instead of `sorted` as you suggest, we need just two memory allocations instead of three.
But my solution is even faster, it needs just a single allocation.
Nevertheless, thanks for the review: I will add a comment about this - it will be useful for anyone who reads the code (including myself in future).